pkg/query/storeset.go

```diff
@@ -215,21 +225,23 @@ func (s *StoreSet) Update(ctx context.Context) {
 			if err != nil {
 				return err
 			}
+
```

Comment:
The code is a bit hard to follow. What about something like:
```
func isUnhealtyh(ctx, s store, unhelathyStores syncmap[addr]struct{}{}) { ... }

// Remove all unhealthy stores
unhealthyStores := synncmap.Map{}
for _, s := range s.stores {
   go(s) { isUnhealthy(ctx, s, &unhealthyStores }
}
wait

for _, unhealthy := range unhealthyStores {
  delete s.stores[unhelathy]
}

// Add new specs to stores
specs  := s.storeSpecs()
newStores := syncmap.Map{}
for _, spec := range specs {
  go(spec) {
    st := storeFromSpec
    if st != nil { newStores[addr] = st }
  }
}
wait
```
and so on. Feel free to ping me on slack for more info.

Reply:
will do

Reply:
refactored `Update` method.